lib/compiler/typescript-loader.ts

```diff
@@ -6,6 +6,12 @@ export class TypeScriptBinaryLoader {
   private tsBinary?: typeof ts;
 
   public load(): typeof ts {
+    if (typeof this.tsBinary !== 'undefined' && 'pnp' in process.versions) {
```

Comment:
Can we move this:

```typescript
typeof this.tsBinary !== 'undefined' && 'pnp' in process.versions
```

Example: `isYarnBerryPnpEnabled()` - this way it wouldn't require an additional comment

into a separate method?